components/layout/query.rs

```diff
@@ -665,14 +668,23 @@ pub(crate) fn process_scroll_parent_query(
 
     // - The element is the root element.
     // - The element is the body element.
-    // - The element’s computed value of the position property is fixed and no ancestor
-    //   establishes a fixed position containing block.
-    if flags.intersects(
-        FragmentFlags::IS_ROOT_ELEMENT | FragmentFlags::IS_BODY_ELEMENT_OF_HTML_ELEMENT_ROOT,
-    ) {
+    //
+    // Note: We only do this for `scrollParent`, as `scrollIntoView` on the `<body>` element should
+    // scroll the body by scrolling the viewport, but the `scrollParent` of the body is `null`. in
+    // `HTMLElement`.
```

Comment:
Also for the root. The trailing "in `HTMLElement`" seems accidental. And "scroll the body by scrolling the viewport" seems confusing.

```suggestion
    // Note: We only do this for `scrollParent`, which needs to be null. But `scrollIntoView` on the
    // `<body>` or root element should still bring it into view by scrolling the viewport.
```